processor/transformprocessor/README.md

```diff
@@ -363,6 +364,51 @@ Examples:
 - `scale_metric(0.1)`: Scale the metric by a factor of `0.1`. The unit of the metric will not be modified.
 - `scale_metric(10.0, "kWh")`: Scale the metric by a factor of `10.0` and sets the unit to `kWh`.
 
+### aggregate_on_attributes
+
+`aggregate_on_attributes(function, Optional[attributes])`
+
+The `aggregate_on_attributes` function aggregates all datapoints in the metric based on the supplied attributes. To have the datapoints aggregated, each datapoint has to contain all of the specified attribute keys. If one of the key is missing in the attributes list, the datapoint will be left out from the aggregation. If no attributes are specified, all attributes of the metric datapoints are taken into consideration for aggregation, which means that every datapoint needs to have the same keys present in its attribute list.
```

Comment:
Something doesn't seem right about this paragraph. It seems to be suggesting that datapoints that have attributes don't have matching attributes will be left alone, but those datapoints will have any non-matching attributes removed and they'll be aggregated together right?

My understanding of the function was that it dropped all attributes that were not in the supplied attributes list and then aggregated the datapoints based on any remaining attributes.  That means that if `attributes` is empty, every attribute is dropped and all datapoints are combined into one.